### PR TITLE
lection_5

### PR DESCRIPTION
**Pull request by VyacheslavReetz**
HW for lection № 5
task 1
task 2
task 3

### CODE REVIEW

File: lection5/task2.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+//   // Пример: ~ 1000
+//   4 548 = 4,5 Kb (Real 4,4 Kb)
+//   454 548 = 454,5 Kb
+//   1 454 548 = 1,5 Mb
+
+//   Результат должен быть округлен до 1 знака после запятой методом .toFixed(), про который надо почитать самим ;)
+
+const sizeInBytes = 45482132131231;
+let result = sizeInBytes;
+let units = 'bytes';
```
Comment (anatoly-karpovich) [inline lection5/task2.js:19]: Где-то я видел этот формат:)))))

File: lection5/task3.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+const str = 'Kalabanga'; //слово
+const vowelsStr = 'aeiouy'; //гласные
+let vowels = 0;
+let consonants = 0;
+
+for (let char of str) {
+    if(vowelsStr.indexOf(char) > -1) {
+        vowels++
+      } else {
+        consonants++
```
Comment (anatoly-karpovich) [inline lection5/task3.js:19]: Word can contain "-" symbol. So need to have consonants string too and check if value is inclued in consonants

File: lection5/task3.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+const alphabet = 'abcdefghijklmnopqrstuvwxyz';
+let result = '';
+
+for (let char of string) {
+    let correctCase = alphabet;
+    if (alphabet.indexOf(char) === -1) {
+        correctCase = alphabet.toUpperCase();
+    }
+    let newIndex = correctCase.indexOf(char) + offset;
+    if (newIndex === -1 || newIndex === 26) {
```
Comment (anatoly-karpovich) [inline lection5/task3.js:41]: Handle offset === 1 too)

**Review by anatoly-karpovich — CHANGES_REQUESTED** [review]

File: lection5/task3.js (hunk 3 above)
Comment (VyacheslavReetz) [reply lection5/task3.js:41]: its working with 1 too)

File: lection5/task2.js (hunk 1 above)
Comment (VyacheslavReetz) [reply lection5/task2.js:19]: не понял намёка)

File: lection5/task3.js (hunk 2 above)
Comment (VyacheslavReetz) [reply lection5/task3.js:19]: done

File: lection5/task3.js (hunk 3 above)
Comment (anatoly-karpovich) [reply lection5/task3.js:41]: my bad, it works)

**Review by anatoly-karpovich — APPROVED** [review]

File: lection5/task3.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+//   Ответ должен выводиться шаблонным литералом вида word contains x vowels and y consonants
+
+const str = 'Kala-banga'; //слово
+const vowelsStr = 'aeiouy'; //гласные
+let vowels = 0;
+let consonants = 0;
+
+for (let char of str) {
+    if (char === '-') {
+        continue;
```
Comment (VladimirLQA) [inline lection5/task3.js:14]: Word  can contain special characters or numbers ...
So need to have consonants string too and check if value is inclued in consonants

**Review by VladimirLQA — APPROVED** [review]

File: lection5/task3.js (hunk 4 above)
Comment (VyacheslavReetz) [reply lection5/task3.js:14]: Thats not in task)
but ok, changed + add check for uppercase values

Comment (VladimirLQA) [reply lection5/task3.js:14]: `Thats not in task)`
The function accepts a `word`, this word, in 99% of cases, is represented as a string, but the string can have a numbers, symbols, so on

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**